dvc/repo/add.py

```diff
@@ -56,21 +65,15 @@ def _find_all_targets(repo, target, recursive):
     return [target]
 
 
-def _create_stages(repo, targets, fname, no_commit):
+def _create_stages(repo, targets, fname):
     stages = []
 
-    with repo.state:
-        for out in targets:
-            stage = Stage.create(repo, outs=[out], add=True, fname=fname)
-
-            if not stage:
-                continue
+    for out in targets:
+        stage = Stage.create(repo, outs=[out], add=True, fname=fname)
 
-            stage.save()
-
-            if not no_commit:
-                stage.commit()
+        if not stage:
```

Comment:
Is this possible? Or is this some code artifact?

Reply:
Possible if already cached.